### DISCUSSION TITLE
Structurizr onpremises uses DSL Editor in restricted mode, why?

### DISCUSSION THREAD

**Discussion by MetalDude101** [structurizr/onpremises#97] [General]
Structurizr onpremises uses DSL Editor in restricted mode, thereby not allowing any DSL to contain !include, !adrs, !docs, and "extends".
Is there a reason for this?

com.structurizr.onpremises.web.workspace.DslEditorController

    private Workspace fromDsl(String dsl) throws StructurizrDslParserException, WorkspaceScopeValidationException {
        StructurizrDslParser parser = new StructurizrDslParser();
		parser.setRestricted(true);		<--- Why ?
        parser.parse(dsl);

It does not make it easy to evaluate using this , if i cannot test multiple workspaces that might reference each other.
There does not seem to be a properties setting to override this.

**Comment by simonbrowndotje** [comment]
> Is there a reason for this?

A typical installation will see the Structurizr data directory locked down and inaccessible to users. Running the DSL in restricted mode prevents the DSL parser (which runs on the server) from (1) accessing files on the server and (2) running untrusted code via scripts/plugins.

**Reply by MetalDude101** [reply]
Can this default behavior be changed ? How can i test "extends" without fully writing a program to implement your API to do what i want to do? Can a properties file setting be added to control this?

**Reply by simonbrowndotje** [reply]
You'll need to use the Structurizr CLI if you want to use `extends`.
